include/swift/AST/DiagnosticEngine.h

```diff
@@ -425,25 +439,87 @@ namespace swift {
     /// Add a character-based range to the currently-active diagnostic.
     InFlightDiagnostic &highlightChars(SourceLoc Start, SourceLoc End);
 
+    static const char *fixItStringFor(const FixItID id);
+
+    /// Add a token-based replacement fix-it to the currently-active
+    /// diagnostic.
+    template <typename... ArgTypes>
+    InFlightDiagnostic &
+    fixItReplace(SourceRange R, StructuredFixIt<ArgTypes...> fixIt,
+                 typename detail::PassArgument<ArgTypes>::type... VArgs) {
+      SmallVector<DiagnosticArgument, 3> Args;
+      DiagnosticArgument DiagArgs[] = {
+        DiagnosticArgument(0), std::move(VArgs)...
+      };
+      Args.append(DiagArgs + 1, DiagArgs + 1 + sizeof...(VArgs));
```

Comment:
I'm actually reading this now and I think you've made this more complicated than necessary. You should be able to pass DiagArgs directly below; ArrayRef has constructors from many kinds of contiguous storage, including fixed-sized arrays.

Reply:
yeah, you're right, the extra step was unnecessary here. Time to spend some time familiarizing myself with the llvm docs!